src/viewer.js

```diff
@@ -2657,8 +2657,8 @@ function onCanvasKeyDown( event ) {
     var canvasKeyDownEventArgs = {
       originalEvent: event.originalEvent,
       preventDefaultAction: false,
-      preventVerticalPan: event.preventVerticalPan,
-      preventHorizontalPan: event.preventHorizontalPan
+      preventVerticalPan: !this.panVertical,
```

Comment:
I don't know for sure if the passed-in `event.preventVerticalPan` is ever used, but just to be on the safe side, I think it would be better to just add the new logic, so this line would become:

`preventVerticalPan: event.preventVerticalPan && !this.panVertical,`

Same for the other three similar lines, of course.

Reply:
I see your point. I did some debugging and discovered that `event.preventVerticalPan` and `event.preventHorizontalPan` always remain undefined and are never updated thus, causing the issue. From my understanding, wouldn't `preventVerticalPan: event.preventVerticalPan && !this.panVertical` always evaulate to undefined? I haven't tested to see if this is the outcome yet but I'm just curious

Reply:
Right, it should be `event.preventVerticalPan || !this.panVertical`... sorry for getting that wrong! Even with `event.preventVerticalPan` being undefined that should work.